common/djangoapps/student/views.py

```diff
@@ -702,8 +702,19 @@ def dashboard(request):
     course_entitlement_available_sessions = {}
     for course_entitlement in course_entitlements:
         course_entitlement.update_expired_at()
-        course_entitlement_available_sessions[str(course_entitlement.uuid)] = \
-            get_course_runs_for_course(str(course_entitlement.course_uuid))
+        # Filter only the course runs that do not have an enrollment_end date set, or have one set in the future
+        course_runs_for_course = get_course_runs_for_course(str(course_entitlement.course_uuid))
+        enrollable_course_runs = []
+
+        for course_run in course_runs_for_course:
+            if 'enrollment_end' not in course_run:
```

Comment:
rather then an `if..else` block here, you could just do `enrollment_end = course_run.get('enrollment_end', None)` and then proceed with the logic under `else`